pkg/controllers/user/nslabels/labels.go

```diff
@@ -70,3 +76,26 @@ func (nsh *namespaceHandler) addProjectIDLabelToNamespace(ns *corev1.Namespace,
 
 	return nil
 }
+
+func (nsh *namespaceHandler) updateProjectIDLabelForSecrets(projectIDFieldValue, projectID string, namespace string) error {
+	if projectIDFieldValue == "" {
+		return nil
+	}
+	secrets, err := nsh.secrets.List(metav1.ListOptions{FieldSelector: fmt.Sprintf("metadata.namespace=%s", namespace)})
+	if err != nil {
+		return err
+	}
+	for _, secret := range secrets.Items {
+		if secret.Annotations[ProjectScopedSecretAnnotation] == "true" {
+			if err := nsh.secrets.DeleteNamespaced(namespace, secret.Name, &metav1.DeleteOptions{}); err != nil {
+				return err
+			}
+		} else if secret.Annotations[ProjectIDFieldLabel] != "" {
```

Comment:
My only question/concern is this:
Are you accidentally changing the value of the field label from `"<clsuterID>:<projectID>"` to `"<projectID>"`?
Because the projectID passed is just the projectID portion of splitting the original `ProjectIDFieldLabel` field

Reply:
Yes, since projectID was the split part, I am actually replacing just that with the new value - 
```
secret.Annotations[ProjectIDFieldLabel] = strings.Replace(secret.Annotations[ProjectIDFieldLabel], projectIDFieldValue, projectID, 1)
``` 
But on second thought, I'll update and pass clusterID to the function instead